ci-operator/step-registry/ipi/install/install/ipi-install-install-ref.yaml

```diff
@@ -28,6 +28,18 @@ ref:
   - name: SPOT_INSTANCES
     default: "false"
     documentation:  "Use AWS Spot Instances for worker nodes.  Set to 'true' to opt into spot instances.  Explicitly set to 'false' to opt out.  Leave unset for the default, which may change."
+  - name: HETEROGENEOUS
+    default: "false"
+    documentation: "Adds additional workers of different architecture."
```

Comment:
This feels a bit redundant.  Or at least, there are multiple levels of knobs:

* `HETEROGENEOUS` is a big, generic knob.  I'd expect setting this true to give you a heterogeneous cluster without having to care about details like "how many heterogeneous compute?" or "which architecture should they run?".
* `ADDITIONAL_WORKER_ARCHITECTURE` is a mid-level knob, which can express more details than `HETEROGENEOUS`, but doesn't have to dip into anything cloud-specific (and also cannot control things like memory available to the heterogeneous nodes).
* `ADDITIONAL_WORKER_VM_TYPE` is a low-level knob, which gives folks full control, but needs cloud-specific knowledge (and possibly CI-account-quota-specific-knowledge) to be able to set it.

I'm not clear on whether we need all three layers of knobs, or how we avoid confusing people who set impossible combinations (e.g. `HETEROGENEOUS=false` while asking for ARM `ADDITIONAL_WORKER_ARCHITECTURE` and setting a s390x `ADDITIONAL_WORKER_VM_TYPE`).

Reply:
and to add to this how would the user add multiple architectures in the future if they wanted to? should heterogeneous have its own configuration object?

Reply:
we could also do away with `ADDITIONAL_WORKER_VM_TYPE` by choosing a sane default ourselves

Reply:
> I'm not clear on whether we need all three layers of knobs
> HETEROGENEOUS=false while asking for ARM ADDITIONAL_WORKER_ARCHITECTURE

`HETEROGENEOUS=false` currently leads to any `ADDITIONAL_*` var being ignored. 

> while asking for ARM ADDITIONAL_WORKER_ARCHITECTURE and setting a s390x 
 ADDITIONAL_WORKER_VM_TYPE

This would fail even for a user launching the installer with a wrong install-config. For instance, the following install-config (for an arm64 installation when the user sets an amd64 instance type) would not produce a validation error, leading the installation to fail later. 

```yaml
compute:
- architecture: arm64
  hyperthreading: Enabled
  name: worker
  platform:
    aws:
      type: m6a.xlarge
  replicas: 3
```

> and to add to this how would the user add multiple architectures in the future if they wanted to? should heterogeneous have its own configuration object?

Could we address this whenever we get the `compute` array in the install-config supporting heterogeneous architectures and multiple values? 

doing this now in this repo may mean that the additional workers are provisioned at a different time than the other original workers provided by the installer via the install-config.yaml. It leads to >15mins increased time before running the tests. By "hacking" the manifests created by the installer, we are actually doing an installation with heterogeneous workers at day-0 and this should be considered temporary IMO, until we get the installer supporting this conf.

> we could also do away with ADDITIONAL_WORKER_VM_TYPE by choosing a sane default ourselves based on arch

I was thinking at a solution like the SIZE_VARIANT we leverage, and also got this recent change that should/could be better addressed for arm64 jobs (currently, we set COMPUTE_NODE_TYPE at job config level rather than having a per-arch default): https://github.com/openshift/release/blob/f5463172571a2f32cda2dd51df3d4569d7f2415b/ci-operator/step-registry/ipi/conf/aws/ipi-conf-aws-commands.sh#L62-L72

IMO, beyond a possible default, we should consider a `sizing` capability for specific testing cases like the performance and scalability ones.

Reply:
we will get a validation error when we specify differing arches for compute and control plane. some of these hacks will go away with IPI support once we grow that organically in the installer but a separate step for heterogeneous still looks like the way to go.